### PR TITLE
Shellcheck everything

### PR DESCRIPTION
**Pull request by ggreer**
- `make shellcheck` now checks shell scripts in `tests/` and  `modules/`
- Update all shell scripts in those dirs to pass, either by fixing the warning/error or disabling the rule for that line.

Shellcheck is part of the Jenkins tests, so new lint errors should fail the build. We can clean up the existing stuff in time. Basically, this stops us from making new mistakes.

### CODE REVIEW

**Comment by ggreer** [comment]
Tests failed because of AWS limits. Shellcheck stuff is fine.

File: modules/aws/ignition/resources/s3-puller.sh [hunk 1]
```diff
@@ -18,4 +19,5 @@ fi
         done
     '
 
+# shellcheck disable=SC1001,SC1083,SC2086
 /usr/bin/sudo mv /tmp/$${1//\//+} $2
```
Comment (kans) [inline modules/aws/ignition/resources/s3-puller.sh]: Does anyone know what this does?

File: modules/update-payload/make-update-payload.sh [hunk 2]
```diff
@@ -45,17 +45,23 @@ echo "Using deployments: [${deployments[*]}]" >&2
...
 for f in ${deployments[*]}; do
   tmpfile=$(mktemp /tmp/deployment.XXXXXX)
+  # shellcheck disable=SC2086
   yaml2json < ${ASSETS_DIR}/${f} > ${tmpfile}
   tmpfiles+=(${tmpfile})
 done
 
+# TODO: (ggreer) I'm pretty sure ">" is *not* what we want here
+# shellcheck disable=SC2071
 if [[ ${#tmpfiles[*]} > 0 ]]; then
```
Comment (kans) [inline modules/update-payload/make-update-payload.sh]: I don't think this is even close.   Lets fix it now?

File: modules/aws/ignition/resources/s3-puller.sh (hunk 1 above)
Comment (ggreer) [inline modules/aws/ignition/resources/s3-puller.sh]: It is a mystery.

File: modules/update-payload/make-update-payload.sh (hunk 2 above)
Comment (ggreer) [inline modules/update-payload/make-update-payload.sh]: I'm worried by "fixing" it, we'll break something. For all we know, this script depends on broken behavior. :(

**Pull request merged**